lib/auth/httpAuthMiddleware.js

```diff
@@ -7,23 +8,64 @@ const { Strategy } = require('./strategy')
 let options
 let passport
 let httpNodeApp
+let client
+const httpTokenCache = {}
 
 module.exports = {
     init (_options) {
         options = _options
-        return (req, res, next) => {
-            try {
-                if (req.session.ffSession) {
-                    next()
-                } else {
-                    req.session.redirectTo = req.originalUrl
-                    passport.authenticate('FlowFuse', { session: false })(req, res, next)
+        return [
+            async (req, res, next) => {
+                try {
+                    if (req.session.ffSession) {
+                        next()
+                    } else if (req.get('Authorization')?.startsWith('Bearer')) {
+                        // need to make a HTTP request against the forge platform to check
+                        // the token. Should we cache the token to reduce round trip on every
+                        // request?
+                        // We should include the Project ID and the path along with the token
+                        // to be checked
+                        const token = req.get('Authorization').split(' ')[1]
+                        const cacheHit = httpTokenCache[token] 
+                        if (cacheHit) {
+                            const age = (Date.now() - cacheHit.age) / 1000
+                            if (age < 300) {
+                                next()
+                                return
+                            }
+                            delete httpTokenCache[token]
+                        }
+                        const query = {
+                            path: req.path
+                        }
+                        try {
+                            const result = await client.get(options.projectId, {
```

Comment:
```suggestion
                            await client.get(options.projectId, {
```

'result' is assigned a value but never used.eslint[no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars)